### PR TITLE
Add support for HMAC-SHA256

### PR DESCRIPTION
**Pull request by mick88**

### CODE REVIEW

**Comment by thedrow** [comment]
Looks really nice but needs test coverage.

File: oauthlib/oauth1/rfc5849/__init__.py [hunk 1]
```diff
@@ -27,10 +27,12 @@
...
 from . import parameters, signature
 
-SIGNATURE_HMAC = "HMAC-SHA1"
+SIGNATURE_HMAC_SHA1 = "HMAC-SHA1"
+SIGNATURE_HMAC_SHA256 = "HMAC-SHA256"
+SIGNATURE_HMAC = SIGNATURE_HMAC_SHA1
 SIGNATURE_RSA = "RSA-SHA1"
 SIGNATURE_PLAINTEXT = "PLAINTEXT"
-SIGNATURE_METHODS = (SIGNATURE_HMAC, SIGNATURE_RSA, SIGNATURE_PLAINTEXT)
+SIGNATURE_METHODS = (SIGNATURE_HMAC_SHA1, SIGNATURE_HMAC_SHA256, SIGNATURE_RSA, SIGNATURE_PLAINTEXT)
```
Comment (thedrow) [inline oauthlib/oauth1/rfc5849/__init__.py]: This will break backwards compatibility. Please include both `SIGNATURE_HMAC_SHA1` & `SIGNATURE_HMAC`.

Comment (kevin-brown) [inline oauthlib/oauth1/rfc5849/__init__.py]: Are those two the exact same? (line 32)

**Comment by ViktorHaag** [comment]
Superseded with PR #498 ?

**Comment by lepture** [comment]
This should not be included in this library. It is not in RFC. However, we can provide a way to extend signature methods.

**Pull request closed without merging**